src/QsCompiler/Compiler/RewriteSteps/IRewriteStepsLoader.cs

```diff
@@ -0,0 +1,14 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Immutable;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.Quantum.QsCompiler
+{
+    internal interface IRewriteStepsLoader
+    {
+        ImmutableArray<LoadedStep> GetLoadedSteps(CompilationLoader.Configuration config);
```

Comment:
The only think that bothers me slightly with this setup is that each step gets the full configuration object; also the ExternalRewriteStepsManager is an IRewriteStepsLoader. It would be sufficient for the manager to get the configuration and then ensure that each loader gets the necessary arguments.

Reply:
sure I will change it, will need to think a bit how to reflect this best

Reply:
done - I removed the interface altogether and simplified everything. hopefully it is more reasonable now 😀